### PR TITLE
LDAP Auth Header & Ports updates

### PR DESCRIPTION
**Pull request by tlblessing**
https://konghq.atlassian.net/browse/DOCS-1269 Shane requested more information and an example for LDAP auth header. This ticket was a subticket of https://konghq.atlassian.net/browse/DOCS-1162.

While I was in there, also did some copy editing and added in missing defaults and examples to match the schema and whatever the basic LDAP plugin has. The copy edits from this PR can be ported to the community LDAP plugin after this one is reviewed and approved.

Direct link main entry point:

https://deploy-preview-2357--kongdocs.netlify.app/hub/kong-inc/ldap-auth-advanced/

Auth header example (added text highlighted in yellow):

image

See https://deploy-preview-2357--kongdocs.netlify.app/hub/kong-inc/ldap-auth-advanced/#usage.

Additional ticket: https://konghq.atlassian.net/browse/DOCS-1348 Added more information about ports and the associated mutually exclusive ports. Please carefully review the updates to the ldap_port, start_tls, and ldps config descriptions. Per the schema, these fields are mutually exclusive:  if ldaps and start_tls then    return nil, "ldaps and StartTLS cannot be enabled simultaneously."

Questions for reviewers: What are the supported protocols? I don't see it in the schema file https://github.com/Kong/kong-plugin-enterprise-ldap-auth/blob/master/kong/plugins/ldap-auth-advanced/schema.lua. The community version of the plugin lists these:

image

### CODE REVIEW

**Comment by darrenjennings** [comment]
> What are the supported protocols?
I believe this is just an oversight on the plugin schema for enterprise. It should support the same protocols as the community edition.

**Review by darrenjennings — APPROVED** [review]

**Comment by tlblessing** [comment]
> > What are the supported protocols?
> 
> I believe this is just an oversight on the plugin schema for enterprise. It should support the same protocols as the community edition.

Thank you @darrenjennings for the additional information and your approval! Will adjust the protocols in  he doc, but someone should probably adjust the schema as well.

File: app/_hub/kong-inc/ldap-auth-advanced/index.md [hunk 1]
```diff
@@ -31,84 +33,92 @@ params:
...
       description: |
-        Cache expiry time in seconds
+        Cache expiry time in seconds.
     - name: timeout
       required: false
       default: "`10000`"
       value_in_examples:
       description: |
-        An optional timeout in milliseconds when waiting for connection with LDAP server
+        An optional timeout in milliseconds when waiting for connection with LDAP server.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ldap-auth-advanced/index.md:105]:
```suggestion
        An optional timeout, in milliseconds, when waiting for connection with LDAP server.
```

File: app/_hub/kong-inc/ldap-auth-advanced/index.md [hunk 2]
```diff
@@ -31,84 +33,92 @@ params:
...
     - name: keepalive
       required: false
       default: "`10000`"
       value_in_examples:
       description: |
-        An optional value in milliseconds that defines for how long an idle connection to LDAP server will live before being closed
+        An optional value in milliseconds that defines how long an idle connection to LDAP server will live before being closed.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ldap-auth-advanced/index.md:111]:
```suggestion
        An optional value, in milliseconds, that defines how long an idle connection to the LDAP server will live before being closed.
```

File: app/_hub/kong-inc/ldap-auth-advanced/index.md [hunk 3]
```diff
@@ -31,84 +33,92 @@ params:
...
     - name: anonymous
       required: false
       default:
       value_in_examples:
       description: |
-        An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` attribute which is internal to Kong, and **not** its `custom_id`.
+        An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`. The value must refer to the Consumer `id` attribute that is internal to Kong; **not** its `custom_id`.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ldap-auth-advanced/index.md:117]:
```suggestion
        An optional string (consumer UUID) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`. The value must refer to the Consumer `id` attribute that is internal to Kong, **not** its `custom_id`.
```

File: app/_hub/kong-inc/ldap-auth-advanced/index.md [hunk 4]
```diff
@@ -122,7 +132,7 @@ params:
       default: '`[ "username", "custom_id" ]`'
       value_in_examples:
       description: |
-        Whether to authenticate consumer based on `username` and/or `custom_id`
+        Whether to authenticate consumer based on `username` and/or `custom_id`.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ldap-auth-advanced/index.md:135]:
```suggestion
        Whether to authenticate Consumers based on `username` and/or `custom_id`.
```

File: app/_hub/kong-inc/ldap-auth-advanced/index.md [hunk 5]
```diff
@@ -134,14 +144,14 @@ params:
...
-        The DN to bind to. Used to perform LDAP search of user. This bind_dn
+        The DN to bind to. Used to perform LDAP search of user. This `bind_dn`
         should have permissions to search for the user being authenticated.
     - name: group_base_dn
       required:
       default: "matches `conf.base_dn`"
       value_in_examples:
       description: |
-        Sets a distinguished name for the entry where LDAP searches for groups begin.
+        Sets a distinguished name (dn) for the entry where LDAP searches for groups begin.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/ldap-auth-advanced/index.md:154]:
```suggestion
        Sets a distinguished name (DN) for the entry where LDAP searches for groups begin.
```

File: app/_hub/kong-inc/ldap-auth-advanced/index.md (hunk 1 above)
Comment (tlblessing) [reply app/_hub/kong-inc/ldap-auth-advanced/index.md:105]: These commas are optional and prefer not to have these speedbumps to the reader's eye.

stet

File: app/_hub/kong-inc/ldap-auth-advanced/index.md (hunk 2 above)
Comment (tlblessing) [reply app/_hub/kong-inc/ldap-auth-advanced/index.md:111]: These commas are optional and prefer not to have these speedbumps to the reader's eye.

stet

**Pull request merged**